Foundation/src/DeflatingStream.cpp

```diff
@@ -25,29 +31,17 @@ DeflatingStreamBuf::DeflatingStreamBuf(std::istream& istr, StreamType type, int
 	_pOstr(0),
 	_eof(false)
 {
-	_zstr.next_in   = 0;
-	_zstr.avail_in  = 0;
-	_zstr.total_in  = 0;
-	_zstr.next_out  = 0;
-	_zstr.avail_out = 0;
-	_zstr.total_out = 0;
-	_zstr.msg       = 0;
-	_zstr.state     = 0;
-	_zstr.zalloc    = Z_NULL;
-	_zstr.zfree     = Z_NULL;
-	_zstr.opaque    = Z_NULL;
-	_zstr.data_type = 0;
-	_zstr.adler     = 0;
-	_zstr.reserved  = 0;
-
-	_buffer = new char[DEFLATE_BUFFER_SIZE];
-
-	int rc = deflateInit2(&_zstr, level, Z_DEFLATED, 15 + (type == STREAM_GZIP ? 16 : 0), 8, Z_DEFAULT_STRATEGY);
+	std::unique_ptr<char[]> buffer(new char[DEFLATE_BUFFER_SIZE]);
+
+	std::unique_ptr<z_stream> pZstr = std::make_unique<z_stream>(z_stream{});
```

Comment:
The argument `z_stream{}` to `make_unique` seems wrong. Parameters for the ctor shall be passed there, probably the ones used before to initialize `_zstr`.

Reply:
Assuming `Z_NULL` isn't something special and is just a `nullptr` (I would have to check), then this does the same thing. `z_stream{}` value-initializes a `z_stream` and `std::make_unique<z_stream>(z_stream{})` calls the move constructor.

Reply:
Thanks, I was not sure if default init of integers sets the values to 0.

Reply:
Default init does not, but value init does. `int i;` is uninitialized, but `int i{};` is 0.